components/amount-buttons.jsx

```diff
@@ -61,15 +66,10 @@ var AmountOtherButton = React.createClass({
 var AmountButtons = React.createClass({
   mixins: [require('react-intl').IntlMixin],
   getInitialState: function() {
-    var amount = this.props.amount;
-    var presets = this.props.presets;
-    var preset = presets.indexOf(amount);
-    var userInputting = false;
-    if (preset < 0 && amount) {
-      userInputting = true;
-    }
```

Comment:
We don't need to do any of the above in getInitialState. If we do it in render instead, we can get the same results.

It also cleans up the use of userInputting. How could user be inputting during getInitialState? Should always default to false.